Automattic/wp-calypso
# Add dropdown for AI teirs

## Proposed Changes

* Show a dropdown in the AI lightbox for the user to select what tier they want
* Default to opening lightbox for AI instead of going to checkout
* Have link to support if they want a larger than the max tier

NOTES
* There is an issue with the dropdown when navigating with the keyboard, the link in the last item is not focusable. This is [a deeper issue with the Dropdown component](https://github.com/Automattic/wp-calypso/blob/trunk/packages/components/src/select-dropdown/index.jsx#L358) where only the list item is focusable. The logic seems complex and out of the scope of this PR to fix. I am not entirely sure if this is worth approaching the issue differently (a note below the price rather than a dropdown item) but this is what I have come up with so far
* The highlight animation around the change in request limits in the includes section will be handled separately as this PR was already large enough

## Testing Instructions

1. Checkout this branch or use the Calypso Green live link and go to `/pricing/`
2. Click on "Get" next to the Jetpack AI product and make sure the lightbox opens up with tiers to select
![image](https://github.com/Automattic/wp-calypso/assets/65001528/944fcbe5-9d36-4a7c-b411-c4d46eac0300)
3. Open up the dropdown and confirm all the tiers are present
![image](https://github.com/Automattic/wp-calypso/assets/65001528/731c7100-ba90-40d4-aec0-711b02eb8c2d)
4. Switch between them to make sure the price as well as the request limit in the includes section gets updated with each tier
![Screenshot 2024-04-30 at 11 41 12 AM](https://github.com/Automattic/wp-calypso/assets/65001528/6ecf2cfe-a159-4737-987d-0985686fb5be)
5. Pick a tier and then select "Proceed to checkout" and make sure the correct tier is in checkout
![image](https://github.com/Automattic/wp-calypso/assets/65001528/9eff1a4b-d165-48a4-ada8-ed937d4bdedb)

## Pre-merge Checklist
